### ISSUE TITLE
[Boss] Mage-Lord Urom (Oculus)

### ISSUE THREAD

**Issue by Dranah** [WoWManiaUK/Redemption#2541]
This guy is broken again and in various ways.
Oculus dungeon (http://www.wowhead.com/the-oculus), the 3rd boss is Mage-Lord Urom (http://www.wowhead.com/npc=27655)
http://www.wowhead.com/spell=50476/summon-menagerie
http://www.wowhead.com/spell=50495/summon-menagerie
http://www.wowhead.com/spell=50496/summon-menagerie
http://www.wowhead.com/spell=51112/teleport
http://www.wowhead.com/spell=51110/empowered-arcane-explosion

How it should work:
- Players should engage him 3 times, on 3 platforms, each time he casts "summon menagerie" that summons a group of adds "menagerie", that need to be defeated so that the group can move on and teleports urom to the next platform.
- Every time he teleports, he must leave combat and once the group of adds die, the players must get off-combat too.
- When all 3 platforms are done, he must teleport to the central ring platform and wait for players to engage him (off combat!).
- There when you engage him, he fights the players on the ring, being tanked as per normal, until he is meant to cast "empowered arcane explosion", when he has to teleport to the center of the nexus until he finishes the cast. He must not evade, must not reset, and must not be able to die (stuck on 1 hp if so), while in the center.
When his cast is finished he teleports back onto the ring platform, where he was being tanked before the spell.

How it works atm:
**TL;DR Most of the time you can't engage or defeat him, and cannot finish the dungeon.**
Details:
- Every time you attempt engaging him on a platform , he fails to summon his adds (he cast the spell but nothing happens), and teleports away staying in-combat (despite nobody having damaged him , since he is unattackable), and leaves the players behind stuck in combat so they cannot move on.
- Sometimes his adds do spawn, quite seldom, but you're still stuck in combat after you defeated them.
- If you do manage to move on (re-entering dungeon every time to drop combat or whatsoever) he will be on the next platform, sometimes casting Empowered arcane explosion at complete random. 
- On the last platform, he is unattackably following your tank while you're fighting adds that spawned out of nowhere after an explosion he was not supposed to even cast, and then he randomly teleports away to the center (not onto the ring where he should go, but in the center-center where he is only meant to be to cast explosion), but he remains unattackable. 
Unattackable means you can target but not engage, and there the dungeon ends.
- Sometimes, rarely, you can get past him by flying a circle on the 3 platforms without dismounting ur drake, and he teleports from platform to platform without you needing to fight adds and then defeat him completely tank and spank on the last platform.
But if you don't get lucky, dungeon is over cuz u won't be able to engage him.

**Comment by fvicaria** [comment]
nothing has changed in months so unlikely that "is broken again"

**Comment by Dranah** [comment]
Some months ago he had an issue with constantly reseting to full health when he teleported to the center to cast explosion. That was a different bug which was fixed and otherwise he worked properly back then.
Now he can hardly be engaged and this problem appeared only after one of the latest updates, though its been weeks ago but i only reported it now because i simply didnt have time to deal with this earlier.
But this is a different problem and it broke with one of the updates.

**Comment by Keader** [comment]
I dont touch in Oculus Val Shits.
I just removed some hacks in OBS.

**Comment by squalllh1989** [comment]
Confirmed that this is bugged again like Dranah say

**Comment by Dranah** [comment]
@Keader that's why i said perhaps in capital, i don't mean to blame you, i blame val. There are bugs constantly coming back which were fixed by him back then. I don't really know much about coding, but if his fixes rely on hacks to work, it's possible that simple updates can mess them up, right?
Either way, i'm a tester, it's not my problem WHY it broke, i can only report that it did.

**Comment by Dranah** [comment]
Ok its different now, but not very good yet.

- He is now attackable on platforms and u have to hit him to engage (which would be fine, if it wasn't for you being able to kill him there)
- Problem is you can interrupt his "summon menagerie" cast and then simply kill him as he stands there idly as it bugs him. 
(The adds are summoned anyway, but even if you restart the encounter, he remains idle and broken by having interrupted his cast previously, so your only option is to kill him as a dummy where he stands.)
Hint: Boss should be immune to stuns, silences and only his "frostbomb" interruptible out of all his spells

- If you DONT stun and nuke him on the first platform, but allow him to teleport, the following 2 scenarios can happen:

1) if you engaged him with a single spell, not touching him afterwards, his teleport is successful, he summons his adds, leaves combat, and so do you once the adds are dead. Fight can go on, rinse and repeat on all platforms.
2) If you engage him with going all-out, he sorta bugs out. His adds are summoned, but your group gets stuck in combat and he casts a 2nd menagerie summon right away as he arrives to the 2nd platform, so he basically skips to the 3rd. I suspect this is caused by having a spell flying towards him in the moment when he teleports away, because it lands after he has teleported, putting him in combat again, and triggering his 2nd platform "summon menagerie" cast, so he teleports straight away to the 3rd platform, 
There needs to be some protection to fly-time spells during his teleport I'd say.

- If you get past the platforms to fight the boss at last, he has the recurring bug of reseting to full health when he teleports to the middle.
- Another bug is you can prevent him from going to the middle by continuously moving while hes teleporting. This will readjust his position to staying with you on the platform as he attempts to follow the tank while hes teleporting.
- Lastly, his empowered arcane explosion is also interruptible.

**Comment by Keader** [comment]
Made some fixes.
Please let me know the results.
Check with janitor when it's up in dev tomorrow.

**Comment by Dranah** [comment]
UPDATE:
- Fixed problem with combat stuck and double port from fly-time spells that land after his teleport
- Fixed problem with summon menagerie being interruptible (i could not silence or stun the boss on platform either)
- Fixed problem with boss not porting to center when his target is kiting him
** Evade bug still persists, he resets to full health as soon as he ports to the middle to cast explosion (this makes the fight still pretty impossible to complete)**
- Fixed problem with his spells being interruptible
**There is a bug with fight not reseting properly if it ends while boss is casting "summon menagerie" on one of his platforms. This can be because the person who pulls does feign death or actually dies to the adds that spawn and oneshot him. 
Reproduce steps: engage boss and die or feign death before his summon cast ends. Engage him again, and he will be idle, standing there for you to kill him.**

**Comment by Keader** [comment]
i cant reproduce this evade problem in my core...
what's the steps?

**Comment by Dranah** [comment]
After all 3 platforms are done, engage boss in central ring. 
Dps him slowly (i was being a fire mage with fireball and pyro), and let him cast teleport.
When he teleports, keep DPSing and your attacks will pop up as "absorb" then "evade" and boss is back to full health in the center.

**Comment by Keader** [comment]
Ready to test again.
Keep in mind that he can make "swiming animation" sometimes, wow mania dont have support to control fly flags correct.
But now he should not drop in middle of air in teleport part

**Comment by Dranah** [comment]
Will test in the evening.

**Comment by Dranah** [comment]
UPDATE:
- Boss now despawn and resets to his first platform no matter when or how the encounter ends. (fixed)
- Boss no longer evades when porting to the center (fixed), explosion works properly. The boss swims backwards instead of teleporting back, which can be a problem because it increases the chance he dies in air.
Current problem is that he is able to die in the center (or in the air while he's swimming back) and his corpse cannot be looted.

**Comment by Keader** [comment]
Will be ready in next restart

**Comment by Dranah** [comment]
UPDATE:
- Boss now properly teleports back to the main tank after his explosion cast is done
- Boss now gets stuck on 1 health if he's about to die in the center
- Boss body is teleported to his starter spot on the inner circle after he dies to avoid any scenarios where he dies mid-air while teleporting
